builds/jazz-build-module/aws-apigateway-module.groovy

```diff
@@ -75,4 +75,63 @@ def injectLambdaIntegration(method, filePath) {
   }
 }
 
+def getApiGatewayCore(accountDetailsPrimary){
+  def gatewayValueInfo
+  for (item in accountDetailsPrimary.REGIONS) {
+    if(item.PRIMARY){
+      gatewayValueInfo = item.API_GATEWAY
+      break
+    }
+  }
+  return getAPIIdNameMapping(gatewayValueInfo.PROD, 'jazz', '*')
+}
+
+def getApigatewayInfo(stage, domain, service, accountDetails, config) {
+	def gatewayValue
+	for (item in accountDetails.REGIONS) {
+		if(item.REGION == config.region){
+			gatewayValue = item.API_GATEWAY
+      break
+		}
+	}
+	if(stage && (stage.endsWith('DEV')) || (stage.endsWith('dev'))) {
+		return getAPIIdNameMapping(gatewayValue.DEV, domain, service)
+	} else if (stage && (stage == 'STG') || (stage == 'stg')) {
+		return getAPIIdNameMapping(gatewayValue.STG, domain, service)
+	} else if (stage && (stage == 'PROD') || (stage == 'prod')) {
+		return getAPIIdNameMapping(gatewayValue.PROD, domain, service)
+	}
+}
+
+def getApigatewayInfoCore(stage, domain, accountDetails) {
```

Comment:
getApiGatewayCore is a method just to return ONLY the api_id for PROD stage in primary account and region. getApiGatewayCore is being used in almost all the build packs for generating the base_url which must have the PROD api_id
But getApigatewayInfoCore is a method to return the api_id for all the stages(PROD, STG and DEV) in primary account and region. getApigatewayInfoCore is being used in service-configuration-data-loader where api_id is being fetched depending on the stages.

Also, line number 114 to 119 are different in the sense that, from 97 to 102, we are sending the servicen_name for mapping but from 114 to 119, we are sending the wildcard *.

Reply:
@SauarvD  Thanks for the clarification. But my point is, even though it is being used in different places, we can avoid that and can able to use one method instead of two, with slight changes.    getApiGatewayCore(accountDetailsPrimary), if you are passing the stage and domain in this method we dont need the other.

Reply:
@SiniWilson removed and replaced getApigatewayCore with getApigatewayInfoCore and made necessary changes